src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +52,122 @@
 public class TranscriptServlet extends HttpServlet {
   private static final String TRANSCRIPT_XML_URL_TEMPLATE =
       "http://video.google.com/timedtext?lang=en&v=";
-  private static final String START_ATTRIBUTE = "start";
-  private static final String DURATION_ATTRIBUTE = "dur";
-  private static final String TEXT_TAG = "text";
-  private static final String TEST_VIDEO_ID = "3ymwOvzhwHs";
+  private static final String ATTRIBUTE_START = "start";
+  private static final String ATTRIBUTE_DURATION = "dur";
+  private static final String TAG_TEXT = "text";
+  private static final String PARAM_LECTURE = "lecture";
+  private static final String PARAM_LECTURE_ID = "id";
+  private static final String PARAM_VIDEO_ID = "video";
+
+  private static DatastoreService datastore;
 
   @Override
   public void init() throws ServletException {
-    // TODO: Implement Transcript.
+    datastore = DatastoreServiceFactory.getDatastoreService();
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: Pass the video ID from another servlet.
-    String transcriptXMLUrl = TRANSCRIPT_XML_URL_TEMPLATE + TEST_VIDEO_ID;
-    DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
-    final DocumentBuilder documentBuilder;
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Document document = getTranscriptXmlAsDocument(request).get();
+    putTranscriptLinesInDatastore(request, document);
+  }
+
+  /**
+   * Gets the transcript for a video as a document. The video to extract the transcript from is
+   * indicated by {@code request}.
```

Comment:
Can you mention that this returns Optional.empty() if there's an error parsing the document?

Reply:
Done!